velox/expression/tests/utils/ArgumentTypeFuzzer.cpp

```diff
@@ -21,20 +21,92 @@
 
 #include "velox/expression/ReverseSignatureBinder.h"
 #include "velox/expression/SignatureBinder.h"
+#include "velox/expression/type_calculation/TypeCalculation.h"
+#include "velox/type/SimpleFunctionApi.h"
 #include "velox/type/Type.h"
 #include "velox/vector/fuzzer/VectorFuzzer.h"
 
 namespace facebook::velox::test {
 
 std::string typeToBaseName(const TypePtr& type) {
-  return boost::algorithm::to_lower_copy(std::string{type->kindName()});
+  return type->isDecimal()
+      ? "decimal"
+      : boost::algorithm::to_lower_copy(std::string{type->kindName()});
 }
 
 std::optional<TypeKind> baseNameToTypeKind(const std::string& typeName) {
   auto kindName = boost::algorithm::to_upper_copy(typeName);
   return tryMapNameToTypeKind(kindName);
 }
 
+void ArgumentTypeFuzzer::determineUnboundedIntegerVariables() {
+  // Assign a random value for all integer values.
+  for (const auto& [variableName, variableInfo] : variables()) {
+    if (!variableInfo.isIntegerParameter() ||
+        integerVariablesBindings_.count(variableName)) {
+      continue;
+    }
+
+    // When variable name is 'precision' and result type is decimal, input type
+    // is regarded the same with the result type.
+    if (variableName == "precision" && returnType_ &&
```

Comment:
This way this logic depends on the variable names is quite fragile. This limitation is not mentioned in the PR description. We may need to come up with a more robust solution.

Reply:
Thanks for noticing. This part is not needed and removed now.
This part was here to solve the issue met on nested greatest and least. With below signature, varargs should have the same precision and scale as the argument type. When a function is nested by greatest only like `greatest(greatest(a, b, c), greatest(e, f, g))`, this part of logic controls the result type of `greatest(e, f, g)` to be the same as that of `greatest(a, b, c)`. But this does not work if other function is nested, e.g. in `greatest(greatest(a, b, c), divide(e, f))`, the result type of divide cannot be controlled. Therefore, I added generateExtremeFunctionArgs to avoid this issue by not allowing nested expressions to be generated as varargs.

```
exec::FunctionSignatureBuilder()
                                .integerVariable("precision")
                                .integerVariable("scale")
                                .returnType("DECIMAL(precision, scale)")
                                .argumentType("DECIMAL(precision, scale)")
                                .variableArity()
                                .build()
```